bravado/mapping/spec.py

```diff
@@ -38,17 +56,47 @@ def build(self):
         validator20.validate_spec(self.spec_dict)
         self.api_url = build_api_serving_url(self.spec_dict, self.origin_url)
         self.definitions = build_models(self.spec_dict['definitions'])
+        self.params = self.build_params()
+        self.responses = self.build_responses()
+        self.resources = build_resources(self)
 
-        # TODO
-        # self.shared_parameters =
-        #   build_parameters(self.spec_dict.get('parameters', {})
+    def build_params(self):
+        params = {}
+        for ref_name, param_spec in self.spec_dict['parameters'].iteritems():
+            # Register under both 'plain' ref name and #/ style ref name
+            params[ref_name] = param_spec
+            params['#/parameters/{0}'.format(ref_name)] = param_spec
+        return params
 
-        # TODO
-        # self.shared_responses =
-        #   build_responses(self.spec_dict.get('responses',{})
+    def build_responses(self):
+        log.warn('TODO: implement Spec::build_responses()')
+        return {}
 
-        self.resources = build_resources(self)
+    def resolve(self, object_spec):
+        """Given an object specification, resolve the $ref if it is a $ref
 
+        :param object_spec: object schema
+        :return: dict
+        :rtype: dict
+        """
+        if len(object_spec) == 1 and '$ref' in object_spec:
```

Comment:
In jsonschema you can have a `$ref` and a `description`, from https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#schema-object- it seems like this is also true for swagger spec.

So this check of len == 1 is going to miss anything with extra fields, right?